src/main/java/org/prebid/server/auction/requestfactory/Ortb2RequestFactory.java

```diff
@@ -217,16 +225,71 @@ public BidRequest enrichBidRequestWithAccountAndPrivacyData(AuctionContext aucti
         final Device device = bidRequest.getDevice();
         final Device enrichedDevice = enrichDevice(device, privacyContext);
 
-        if (enrichedRequestExt != null || enrichedDevice != null) {
+        final Regs regs = bidRequest.getRegs();
+        final Regs enrichedRegs = enrichRegs(regs, privacyContext, account);
+
+        if (enrichedRequestExt != null || enrichedDevice != null || enrichedRegs != null) {
             return bidRequest.toBuilder()
                     .ext(ObjectUtils.defaultIfNull(enrichedRequestExt, requestExt))
                     .device(ObjectUtils.defaultIfNull(enrichedDevice, device))
+                    .regs(ObjectUtils.defaultIfNull(enrichedRegs, regs))
                     .build();
         }
 
         return bidRequest;
     }
 
+    private Regs enrichRegs(Regs regs, PrivacyContext privacyContext, Account account) {
+        final ExtRegs regsExt = regs != null ? regs.getExt() : null;
+        final ExtRegsDsa regsExtDsa = regsExt != null ? regsExt.getDsa() : null;
+        if (regsExtDsa != null) {
+            return null;
+        }
+
+        final AccountDsaConfig accountDsaConfig = Optional.ofNullable(account)
+                .map(Account::getPrivacy)
+                .map(AccountPrivacyConfig::getDsa)
+                .orElse(null);
+        final DefaultDsa defaultDsa = accountDsaConfig != null ? accountDsaConfig.getDefaultDsa() : null;
+        if (defaultDsa == null) {
+            return null;
+        }
+
+        final boolean isGdprOnly = BooleanUtils.isTrue(accountDsaConfig.getGdprOnly());
+        if (!isGdprOnly || privacyContext.getTcfContext().isInGdprScope()) {
+            return Optional.ofNullable(regs)
+                    .map(Regs::toBuilder)
+                    .orElseGet(Regs::builder)
+                    .ext(mapRegsExtDsa(defaultDsa, regsExt))
+                    .build();
+        }
+
+        return regs;
+    }
+
+    private static ExtRegs mapRegsExtDsa(DefaultDsa defaultDsa, ExtRegs regsExt) {
+        final List<ExtRegsDsaTransparency> enrichedDsaTransparencies = defaultDsa.getTransparency()
+                .stream()
+                .map(dsaTransparency -> ExtRegsDsaTransparency.of(dsaTransparency.getDomain(),
+                        dsaTransparency.getParams()))
+                .toList();
+        final ExtRegsDsa enrichedRegsExtDsa = ExtRegsDsa.of(defaultDsa.getRequired(),
+                defaultDsa.getPubrender(),
+                defaultDsa.getDatatopub(),
+                enrichedDsaTransparencies);
+        if (regsExt == null) {
+            return ExtRegs.of(null,
+                    null,
+                    null,
+                    enrichedRegsExtDsa);
+        } else {
```

Comment:
`else` here looks redundant, i'd rather remove it or use the ternary operator

Reply:
I made it that way at first, but it looked bad. This one looks ugly too. But yeah, if both of us liked the ternary flavor better, then be it.